web_app/order_books/haiko/main.py

```diff
@@ -0,0 +1,246 @@
+import asyncio
+from decimal import Decimal
+
+from web_app.order_books.abstractions import OrderBookBase
+from web_app.order_books.constants import TOKEN_MAPPING
+from web_app.order_books.haiko.api_connector import HaikoAPIConnector, HaikoBlastAPIConnector
+from web_app.order_books.haiko.histogram import Histogram
+from web_app.order_books.haiko.logger import get_logger
+
+
+class HaikoOrderBook(OrderBookBase):
+    DEX = "Haiko"
+
+    def __init__(self, token_a, token_b, apply_filtering: bool = False):
+        """
+        Initialize the HaikoOrderBook object.
+        :param token_a: baseToken hexadecimal address
+        :param token_b: quoteToken hexadecimal address
+        :param apply_filtering: bool - If True apply min and max price filtering to the order book data
+        """
+        super().__init__(token_a, token_b)
+        self.haiko_connector = HaikoAPIConnector()
+        self.blast_connector = HaikoBlastAPIConnector()
+        self.apply_filtering = apply_filtering
+        self.logger = get_logger("./logs", echo=True)
+
+        self.token_a_name = TOKEN_MAPPING.get(token_a).name
+        self.token_b_name = TOKEN_MAPPING.get(token_b).name
+        self.token_a_price = Decimal(0)
+        self.token_b_price = Decimal(0)
+
+        self._decimals_diff = 10 ** (self.token_a_decimal - self.token_b_decimal or self.token_a_decimal)
+        self._check_tokens_supported()
+        self._set_usd_prices()
+
+    def _set_usd_prices(self) -> None:
+        """Set USD prices for tokens based on Haiko API."""
+        prices = self.haiko_connector.get_usd_prices(self.token_a_name, self.token_b_name)
+        self.token_a_price = Decimal(prices.get(self.token_a_name, 0))
+        self.token_b_price = Decimal(prices.get(self.token_b_name, 0))
+        if self.token_a_price == 0 or self.token_b_price == 0:
+            raise RuntimeError("Prices for tokens aren't available.")
+
+    def _check_tokens_supported(self) -> None:
```

Comment:
If I initialize the class with token a = `0x07e2c010c0b381f347926d5a203da0335ef17aefee75a89292ef2b0f94924864` (wstETH) and token b = `0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7` it raises `ValueError: One of tokens isn't supported by Haiko
` even though the pair is visible in the app.